### PR TITLE
Allow exiting validators to attest

### LINKED ISSUE #7745: Validator Does Not Attest For EXITING Accounts
# 🐞 Bug Report

### Description

 validator ( v1.0.0-beta.1 ) does not attest for accounts in EXITING status

### Has this worked before in a previous version?

 N.A.

## 🔬 Minimal Reproduction

Voluntary exit an account, _then_ start its validator

NOTE: if the account changes status to EXITING _while_ its validator is running, validator would keep on attesting

## 🔥 Error

<pre><code>
Nov 07 22:24:10 * validator[45603]: time="2020-11-07 22:24:10" level=info msg="Validator status" activationEpoch=0 index=10040 prefix=validator pubKey=0x8510d8b931b4 status=EXITING
</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
Ubuntu 20.04
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
v1.0.0-beta.1
  </code>
</pre>

**Anything else relevant (validator index / public key)?**

EXITING account --

https://beaconscan.com/validator/10040

### PR DESCRIPTION
**Pull request by potuz**
When launching the validator client, if the validator is exiting (in the queue) this PR allows for it to continue attesting. 

Fixes #7745 

Note: not tested, that's why it is a draft still.

### CODE REVIEW

**Comment by terencechain** [comment]
@jerryji can you help test this? Thanks

**Comment by potuz** [comment]
@terencechain It seems it was tested by people in r/ethstaker Discord 
https://discord.com/channels/694822223575384095/774220396722913290

It works as intended, but it would be nice to have a log at the beginning saying that the validator is EXITING. Currently all I can do is just inform this fact. But it would be nice to have the API call message `ValidatorStatusResponse` to actually include a position in the exit queue or at least the exit epoch to inform the user. I think it's better to keep this PR to the point that solves the impending issue, and perhaps I should open a different issue about the API call.

**Comment by terencechain** [comment]
> @terencechain It seems it was tested by people in r/ethstaker Discord
> https://discord.com/channels/694822223575384095/774220396722913290

Great, can you add a regression test? It'll be ready after test

> It works as intended, but it would be nice to have a log at the beginning saying that the validator is EXITING. Currently all I can do is just inform this fact. But it would be nice to have the API call message `ValidatorStatusResponse` to actually include a position in the exit queue or at least the exit epoch to inform the user. I think it's better to keep this PR to the point that solves the impending issue, and perhaps I should open a different issue about the API call.

I'd make the request in the official eth2api repo https://github.com/ethereum/eth2.0-APIs. We should avoid making ethereumapi features at this point since it's going to get deprecated in favor of the official eth2.0 api sometime in the future

**Comment by potuz** [comment]
Added the regression test, I am not comfortable having not tested this at runtime myself, perhaps we could wait for another test of someone with an exiting validator?

File: validator/client/validator_test.go [hunk 1]
```diff
@@ -401,6 +401,40 @@ func TestWaitActivation_LogsActivationEpochOK(t *testing.T) {
...
+		gomock.Any(),
+		&ethpb.ValidatorActivationRequest{
+			PublicKeys: [][]byte{pubKey[:]},
+		},
+	).Return(clientStream, nil)
+	clientStream.EXPECT().Recv().Return(
+		resp,
+		nil,
+	)
+	require.NoError(t, v.WaitForActivation(context.Background()))
```
Comment (terencechain) [inline validator/client/validator_test.go:435]: Shouldn't the test check for `checkAndLogValidatorStatus` and `validatorActivated` returns true?

Comment (potuz) [reply validator/client/validator_test.go:435]: No, this test fails current master without the fix. I checked this before submitting. The test is the minimal reproduction of the actual error: namely `WaitForActivation` does not return `nil` as it should when the validators are EXITING. The reason and the fix are in the function `checkAndLogValidatorStatus` but the bug itself is realized in the runner when calling `WaitForActivation` so I decided to include the minimal reproduction of this bug in this test. Let me know if you'd rather move this to `checkAndLogValidatorStatus` though.

**Review by drumenbassik — APPROVED** [review]

**Comment by potuz** [comment]
@dv8silencer checked it on runtime https://discord.com/channels/476244492043812875/667761857209040916/775406870768123915

File: validator/client/validator_test.go (hunk 1 above)
Comment (terencechain) [reply validator/client/validator_test.go:435]: that makes sense, thanks!

**Review by terencechain — APPROVED** [review]

**Comment by prestonvanloon** [comment]
This might help with #7707

**Pull request merged**